cell_classification/segmentation_data_prep.py

```diff
@@ -119,17 +143,65 @@ def marker_activity_mask(self, instance_mask, cell_types, marker_activity):
             np.array:
                 The marker activity mask
         """
-        pass
+        return np.zeros([500, 500])
 
-    def make_tf_record(self, tf_record_path):
-        """Iterates through the data_folders and loads, transforms and serializes a
-            tfrecord example for each data_folder
+    def prepare_example(self, data_folder, marker):
+        """Prepares a tfrecord example for the given data_folder and marker
+        Args:
+            data_folder (str):
+                The path to the data_folder
+            marker (str):
+                The marker shown in the image, e.g. "CD8" corresponds
+                to file name "CD8.tiff" in the data_folder
+            normalization_dict (dict):
+                The normalization dict
+        Returns:
+            dict:
+                Example dict
+        """
+        # load and normalize the multiplexed image and masks
+        mplex_img = self.get_image(data_folder, marker)
+        mplex_img /= self.normalization_dict[marker]
+        binary_mask, instance_mask = self.get_instance_masks(
+            data_folder, self.cell_mask_key
+        )
+        # get the cell types and marker activity mask
+        cell_types = self.get_cell_types(instance_mask, self.sample_key)
```

Comment:
I think the `get_cell_types` function will need to return a df, not just a list. 

If it just returns a list, let's say `[CD8T, CD4T, CD4T, Tumor]`, and then we get the marker activity for CD8 `[1, 0, 0, -1]`, in order to generate the `marker_activity_mask`, we need to know the integer label of each cell, not just their cell type. That's how we can map back to the instance mask. 

Therefore, something like pandas df with both the cell_type and cell_ID column should probably be returned. Thsi would involve just subsetting the `cell_table` to those columns.

Reply:
I agree.